### PR TITLE
feat: simplify notification emails for better scannability (#362)

### LINKED ISSUE #362: Notification email is too verbose
Too much fluff, formality and so on. Keep it simple, concise and easy to scan.

### PR DESCRIPTION
**Pull request by mnbf9rca**
## Summary
Simplified notification emails by removing fluff and putting key info in subject lines for easier scanning.

### Subject Lines
- **Disruption alerts**: `⚠️ {route_name}: {line1}, {line2}, ... disrupted`
- **Status updates**: `✅ {route_name}: {line1}, {line2} restored`
- **Mixed**: `✅ {route_name}: Victoria restored | Northern still disrupted`

Examples:
- `⚠️ To Work: Elizabeth, Victoria, DLR, Northern, Piccadilly disrupted`
- `✅ Morning Commute: Victoria restored`

### Template Simplifications
**Removed:**
- Greeting ("Hello John,")
- Intro paragraphs ("We're writing to inform you...", "Good news!...")
- Advice ("Please plan your journey accordingly...")
- Preferences text ("This alert was sent based on...")
- Footer copyright block

**Kept:**
- Route name header
- Line status boxes (the actual useful content)
- TfL link (simplified to text link)

### Files Modified
1. `backend/app/services/notification_service.py` - Subject line helpers, simplified plain text
2. `backend/app/templates/email/disruption_alert.html` - Removed verbose content
3. `backend/app/templates/email/status_update.html` - Removed verbose content
4. `backend/tests/test_notification_service.py` - Added 6 new tests, updated assertions

## Test Coverage
- ✅ All 33 tests passing
- ✅ 98.92% coverage on notification_service.py (exceeds 95% requirement)
- ✅ Added tests for subject line generation (single/multiple lines, mixed states)
- ✅ Updated existing test assertions for new subject format

## Test Plan
- [x] All notification service tests pass
- [x] Pre-commit hooks pass
- [x] Coverage exceeds 95% for modified code

Closes #362

## Summary by Sourcery

Simplify notification emails by focusing on concise, informative subject lines and minimal in-body content while keeping key status details.

New Features:
- Introduce helper functions to generate concise disruption and status update email subject lines based on affected and restored lines.

Enhancements:
- Simplify disruption and status update email templates by removing greetings, explanatory copy, preference/footer text, and CTA buttons, leaving just route details, line status boxes, and a compact TfL status link.
- Streamline plain-text email bodies to remove boilerplate and keep only per-line statuses plus a short TfL status URL line.

Tests:
- Add and update tests to cover the new subject line formats and helper functions, including empty, single-line, multi-line, and mixed-status scenarios, and adjust template expectations for the removed greeting.

### CODE REVIEW

File: backend/app/services/notification_service.py [hunk 1]
```diff
@@ -18,6 +18,62 @@
...
+    cleared_names = ", ".join([c.line_name for c in cleared_lines])
+
+    if not still_disrupted:
+        # All clear
+        if len(cleared_lines) == 1:
+            return f"✅ {route_name}: {cleared_names} restored"
+        return f"✅ {route_name}: All clear"
+    # Some still disrupted
+    disrupted_names = ", ".join([d.line_name for d in still_disrupted])
+    return f"✅ {route_name}: {cleared_names} restored | {disrupted_names} still disrupted"
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/app/services/notification_service.py]: Fixed. Modified `_build_status_update_subject` to handle the case where `cleared_lines` is empty but `still_disrupted` is non-empty. Now returns `⚠️ {route_name}: {disrupted} still disrupted` (without the misleading checkmark). See lines 67-70 in notification_service.py.

File: backend/tests/test_notification_service.py [hunk 2]
```diff
@@ -263,8 +267,9 @@ async def test_render_email_template_without_user_name(self) -> None:
             },
         )
 
-        # Verify "there" appears as default greeting
-        assert "Hello there" in html or "there" in html.lower()
+        # Verify template renders correctly without greeting
+        assert "Test Route" in html
+        assert "Victoria" in html
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_notification_service.py:272]: Fixed. Added negative assertions to guard against regressions. See lines 274-278 in test_notification_service.py.

File: backend/tests/test_notification_service.py [hunk 3]
```diff
@@ -910,3 +915,133 @@ async def test_send_status_update_sms_error_propagates(self) -> None:
...
+                line_name="Northern",
+                mode="tube",
+                status_severity=6,
+                status_severity_description="Minor Delays",
+                reason="Signal failure",
+            )
+        ]
+
+        subject = _build_status_update_subject("To Work", cleared_lines, still_disrupted)
+        assert subject == "✅ To Work: Victoria restored | Northern still disrupted"
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/tests/test_notification_service.py]: Fixed. Removed `@pytest.mark.asyncio` decorator and changed `async def` to `def` for all 8 synchronous subject line helper tests.

File: backend/app/services/notification_service.py [hunk 4]
```diff
@@ -18,6 +18,62 @@
...
+
+    Args:
+        route_name: Name of the route affected
+        disruptions: List of disruptions affecting the route
+
+    Returns:
+        Formatted subject line
+    """
+    if not disruptions:
+        return f"⚠️ {route_name}: All clear"
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/app/services/notification_service.py]: Fixed. Changed to raise ValueError instead of returning "All clear" for empty disruptions list. This makes the error explicit and prevents misuse. See lines 35-36 in notification_service.py.

File: backend/app/services/notification_service.py [hunk 5]
```diff
@@ -18,6 +18,62 @@
...
+
+    line_names = ", ".join([d.line_name for d in disruptions])
+
+    if len(disruptions) == 1:
+        return f"⚠️ {route_name}: {line_names} disrupted"
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply backend/app/services/notification_service.py]: Fixed. Removed the redundant conditional - both branches returned identical strings. Now uses a single return statement (line 39 in notification_service.py).

**Comment by mnbf9rca** [comment]
All review comments have been addressed in commit af4def8:

**sourcery-ai comments:**
- ✅ Edge case with empty `cleared_lines`: Fixed in `_build_status_update_subject` (lines 67-70)
- ✅ Negative assertions for greeting removal: Added to tests (lines 274-278) 
- ✅ Additional edge case tests: Added tests for empty cleared_lines + still_disrupted, and both empty
- ✅ Redundant conditional in `_build_disruption_subject`: Removed (simplified to single return)

**copilot comments:**
- ✅ Async decorators removed: Changed 8 synchronous tests from `async def` to `def`
- ✅ Empty disruptions error: Now raises `ValueError` instead of returning "All clear" (lines 35-36)
- ✅ Both lists empty error: Raises `ValueError` when both `cleared_lines` and `still_disrupted` are empty (lines 65-66)

**Testing:**
- ✅ All 35 tests passing
- ✅ Coverage: 98.97% for notification_service.py (exceeds 95% requirement)
- ✅ Pre-commit hooks passing

Ready for review.
